lib/virt_autotest/utils.pm

```diff
@@ -679,11 +679,10 @@ sub subscribe_extensions_and_modules {
 
     $ret = 0;
     my @to_be_subscribed = split(/ /, $args{reg_exts});
-    my $version_id = version_utils::get_version_id(dst_machine => "$args{dst_machine}");
     foreach (@to_be_subscribed) {
-        $cmd = "-p $_/" . $version_id . "/" . get_required_var("ARCH");
+        $cmd = "-p " . "\$(SUSEConnect -l | grep -o \"\\b$_\\/.*\\/.*\\b\")";
```

Comment:
Can you run this once outside of the loop instead?

Reply:
1. If you want to de/register multiple modules/extensions at the same time by using one SUSEConnect command, that does not work. I tried this for you.
2. For multiple modules/extensions, there is need to loop over the array contains them to determine version/arch
3. I tried to execute outside of loop with only one module/extension to be registered, it works as well:
```
my @to_be_subscribed = split(/ /, $args{reg_exts});
$cmd = "-p " . "\$(SUSEConnect -l | grep -o \"\\b@to_be_subscribed\\/.*\\/.*\\b\")";
```
Verification run is [here](http://10.67.131.12/tests/853#step/prepare_transactional_server/8)

Reply:
Sorry, I only meant the `SUSEConnect -l | grep -o \"\\b$_\\/.*\\/.*\\b\"` part to get the product identifier.

Reply:
It has been verified also.